pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -232,6 +240,63 @@ func ApplyShcSecret(mgr *searchHeadClusterPodManager, replicas int32, mock bool)
 				mgr.cr.Status.ShcSecretChanged = append(mgr.cr.Status.ShcSecretChanged, true)
 			}
 		}
+
+		// If admin secret is different from namespace scoped secret change it
+		if adminPwd != nsAdminSecret {
+			scopedLog.Info("admin password different from namespace scoped secret, changing admin password")
+			// If admin password already changed, ignore
+			if i < int32(len(mgr.cr.Status.AdminSecretChanged)) {
+				if mgr.cr.Status.AdminSecretChanged[i] {
+					continue
+				}
+			}
+
+			// Change admin password on splunk instance of pod
+			command := fmt.Sprintf("/opt/splunk/bin/splunk cmd splunkd rest --noauth POST /services/admin/users/admin 'password=%s'", nsAdminSecret)
+			_, _, err = splutil.PodExecCommand(mgr.c, shPodName, mgr.cr.GetNamespace(), []string{"/bin/sh"}, command, false, false)
+			if err != nil {
+				if !mock {
+					return err
+				}
+			}
+			scopedLog.Info("admin password changed on the splunk instance of pod")
+
+			// Get client for Pod and restart splunk instance on pod
+			shClient := mgr.getClient(i)
+			err = shClient.RestartSplunk()
+			if err != nil {
+				return err
+			}
+			scopedLog.Info("Restarted Splunk")
+
+			// Set the adminSecretChanged changed flag to true
+			if i < int32(len(mgr.cr.Status.AdminSecretChanged)) {
+				mgr.cr.Status.AdminSecretChanged[i] = true
+			} else {
+				scopedLog.Info("Appending to AdminSecretChanged")
+				mgr.cr.Status.AdminSecretChanged = append(mgr.cr.Status.AdminSecretChanged, true)
```

Comment:
do we need to explicitly call `append` ? I think instead of this if/else you can just simply do `mgr.cr.Status.AdminSecretChanged[i] = true`, isn't it?

Reply:
Nit: In your example, you are using the pointer directly. Actually, go lang append API takes care of 'Nil' case, so that is not an issue.

Reply:
Ok, in that case you can just use `mgr.cr.Status.AdminSecretChanged = append(mgr.cr.Status.AdminSecretChanged, true)` and can get rid of if/else part.

Reply:
That will not work either because if we have populated at an index **i** already then we shouldn't append another entry for it. We need the index of the slice to match the pod number.

Reply:
Oh okay I get it now!! Sorry for the confusion 👍